File: testdata/dataflow_sample.go
package testdata

import (
	"encoding/json"
	"io"
	"net/http"
)

func HandleRequest(w http.ResponseWriter, r *http.Request) {
	body, _ := io.ReadAll(r.Body)
	var review map[string]interface{}
	json.Unmarshal(body, &review)
	name := review["name"]
	query := "SELECT * FROM users WHERE name = " + name.(string)
	_ = query
}

func SimpleAssignment() string {
	x := "hello"
	y := x
	return y
}

func MultiAssignment() {
	a, b := 1, 2
	_ = a
	_ = b
}

func FieldAccess() {
	r := &http.Request{}
	host := r.Host
	_ = host
}

func ChainedFieldAccess(r *http.Request) string {
	return r.URL.Path
}

File: testdata/dataflow_sample.rs
use std::io::Read;

fn handle_request(request: &str) -> String {
    let body = request.to_string();
    let payload: serde_json::Value = serde_json::from_str(&body).unwrap();
    let name = payload["name"].as_str().unwrap();
    let query = format!("SELECT * FROM users WHERE name = {}", name);
    query
}

fn simple_assignment() -> String {
    let x = String::from("hello");
    let y = x;
    return y;
}

fn field_access(user: &User) -> String {
    let email = user.email.clone();
    email
}

struct User {
    email: String,
}
